mlflow/pyfunc/stdin_server.py

```diff
@@ -0,0 +1,40 @@
+import sys
+import argparse
+import json
+import mlflow
+from mlflow.pyfunc import scoring_server
+import logging
+from pathlib import Path
+
+_logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--model-uri")
+args = parser.parse_args()
+
+_logger.info("Loading model from %s", args.model_uri)
+model = mlflow.pyfunc.load_model(args.model_uri)
+input_schema = model.metadata.get_input_schema()
+_logger.info("Loaded model")
+
+_logger.info("Waiting for request")
+for line in sys.stdin:
+    _logger.info("Received request")
+    request = json.loads(line)
+    data = request["data"]
+    output_file = Path(request["output_file"])
+
+    _logger.info("Parsing data")
+    data = scoring_server.infer_and_parse_json_input(data, input_schema)
+
+    _logger.info("Making predictions")
+    preds = model.predict(data)
+
+    _logger.info("Writing predictions to %s", output_file)
+    with Path(output_file).open("w") as f:
```

Comment:
Why not directly use `stdout` ?

Reply:
Oh I got it, the output content might be mixed with other logging output

Reply:
As I explained in the standup, the model might log in stdout.

Reply:
Discussion:

Can we use `os.mkfifo(path)` instead ? Then we don't need to clean the temp file.

Reply:
Does `os.mkfifo(path)` require clean up?

https://docs.python.org/3/library/os.html#os.mkfifo says:

> FIFOs exist until they are deleted (for example with [os.unlink()](https://docs.python.org/3/library/os.html#os.unlink))

Reply:
But fifo saves disk space.
if you output to file, the file contains all requests outputs until we delete it. But we don't need to save them

Reply:
Makes sense. I'll make the change.

Reply:
I'll test input containing special characters.